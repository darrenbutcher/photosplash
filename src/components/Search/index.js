import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Form } from 'semantic-ui-react'

const SearchForm = styled(Form)`
`

const Search = (props) => {
  const handleSearchSubmit = (e, onSubmit) => {
    e.preventDefault()
    onSubmit(e)
  }

  return (
    <SearchForm onSubmit={(e) => handleSearchSubmit(e, props.onSubmit)}>
      <Form.Group>
        <Form.Input
          value={props.value}
          onChange={props.onChange}
          placeholder={props.placeholder}
          type="text"
        />
       <Form.Button>{props.buttonText}</Form.Button>
      </Form.Group>
    </SearchForm>
  )
}

Search.defaultProps = {
  buttonText: 'Search'
}

Search.propTypes = {
  placeholder: PropTypes.string,
  buttonText: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
}

export default Search
