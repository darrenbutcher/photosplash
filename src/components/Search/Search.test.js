import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Search from '.'

describe('Search', () => {
  const requiredProps = {
    onSubmit: jest.fn(),
    onChange: jest.fn(),
  }

  const preventDefault = jest.fn()

  const wrapper = shallow(
    <Search
      {...requiredProps} 
      />
  )

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should prevent default', () => {
    const form = wrapper.dive().find('Form')
    form.simulate('submit', { preventDefault })
    expect(preventDefault).toBeCalled()
  })
})
