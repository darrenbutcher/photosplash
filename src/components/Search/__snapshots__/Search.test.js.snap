// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render properly 1`] = `
<Styled(Form)
  onSubmit={[Function]}
>
  <FormGroup>
    <FormInput
      as={[Function]}
      control={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <FormButton
      as={[Function]}
      control={[Function]}
    >
      Search
    </FormButton>
  </FormGroup>
</Styled(Form)>
`;
